classification_fs_all_cellprofiler_all_feat_perfect.py

```diff
@@ -0,0 +1,329 @@
+
+
+# In[4]:
+
+
+data_dir = pathlib.Path("nf1_data_repo/4_processing_features/data/Plate1/CellProfiler")
+data_comp = data_dir / morph_features
+
+
+# In[5]:
+
+
+# If the file isn't found in the path above then raise an error.
+if not data_comp.is_file():
+    raise FileNotFoundError(f"File '{data_comp}' does not exist")
+    
+morph_feat = pd.read_csv(data_comp)
+
+
+# # Creating binary labels for the genotype
+
+# In[6]:
+
+
+lb = LabelBinarizer()
+morph_feat['genotype_label'] = lb.fit_transform(morph_feat['Metadata_genotype'])
+featdf = morph_feat # Create a copy for modification
+
+
+# # Selecting Features
+
+# In[7]:
+
+
+feat_col = [col for col in featdf.columns if 'Metadata' not in col] # Select all columns that don't contain the Metadata in their name
+
+featdf = featdf[feat_col]
+featdf = featdf.loc[:,featdf.columns != 'Unnamed: 0'] # Remove the unnamed column
+
+
+# # Sampling
+
+# In[8]:
+
+
+def down_sample(df, samp_size):
+    """
+    Parameters
+    ----------
+    df: Pandas Dataframe
+        The dataframe to be sampled.
+    samp_size
+        The sample size to be sampled from each class.
+
+    Returns
+    -------
+    Pandas Dataframe
+        The sampled dataframe.
+
+    """
+    return df.sample(n=samp_size, random_state=rnd_val)
+
+min_class_samps_size = min(featdf['genotype_label'].value_counts().values) # Sample size for the minority class
+dsamp_featdf = featdf.groupby('genotype_label', group_keys=False).apply(down_sample, min_class_samps_size) # Down-sample features according to minority class
+traindf, testdf = train_test_split(dsamp_featdf, random_state=rnd_val, shuffle=True, train_size=0.9) # Train and Test Dataframes
+
+
+# In[9]:
+
+
+train_feat = traindf.to_numpy()
+test_feat = testdf.to_numpy()
+
+
+# # Principle Component Analysis (PCA) Visualization
+
+# In[10]:
+
+
+null_idx = np.nonzero(train_feat[:,-1] == 0)[0]
+wt_idx = np.nonzero(train_feat[:,-1] != 0)[0]
+
+
+# In[11]:
+
+
+pca = PCA(n_components=2)
+pca_features = pca.fit_transform(train_feat[:,0:-1])
+print(f'Explained variance in PC1 and PC2 = {np.sum(pca.explained_variance_ratio_)}')
+null = plt.scatter(pca_features[null_idx,0],pca_features[null_idx,1], marker='x', color='r')
+wt = plt.scatter(pca_features[wt_idx,0],pca_features[wt_idx,1], marker='.', color='b')
+plt.xlabel('PC1')
+plt.ylabel('PC2')
+plt.legend((null, wt), ('null', 'wt'))
+plt.show()
+
+
+# In[12]:
+
+
+reducer = umap.UMAP(random_state=rnd_val)
+reducer.fit(train_feat[:,0:-1])
+
+
+# In[13]:
+
+
+null = plt.scatter(reducer.embedding_[null_idx,0],reducer.embedding_[null_idx,1], marker='x', color='r')
+wt = plt.scatter(reducer.embedding_[wt_idx,0],reducer.embedding_[wt_idx,1], marker='.', color='b')
+plt.title('Embedding of the training set by UMAP', fontsize=24)
+plt.legend((null, wt), ('null', 'wt'), loc='lower right')
+plt.show()
+
+
+# # K Cross Validation
+
+# In[14]:
+
+
+num_splits = 5 # Default number of splits
+
+def kcross_val(model, feat, splits=num_splits):
```

Comment:
Same comment as above to move into utils

Reply:
I'll add that in a future pr I think